src/ray/common/task/task_spec.h

```diff
@@ -34,22 +35,28 @@ typedef int SchedulingClass;
 
 struct SchedulingClassDescriptor {
  public:
-  explicit SchedulingClassDescriptor(ResourceSet rs, FunctionDescriptor fd, int64_t d)
+  explicit SchedulingClassDescriptor(ResourceSet rs,
+                                     FunctionDescriptor fd,
+                                     int64_t d,
+                                     std ::pair<NodeID, bool> node_scheduling_strategy)
       : resource_set(std::move(rs)), function_descriptor(std::move(fd)), depth(d) {}
   ResourceSet resource_set;
   FunctionDescriptor function_descriptor;
   int64_t depth;
+  std::pair<NodeID, bool> node_scheduling_strategy;
```

Comment:
why do we only store node_scheduling_strategy here?

Reply:
Tasks are in the same SchedulingClass if scheduler will make the same decision for each task in the group (e.g. if one task is infeasible to run, all the other tasks will be infeasible as well). Because of this, tasks with different node scheduling strategy should be in the different scheduling classes

Reply:
Changed to store the entire scheduling_strategy.